File: csharp/sieve/Sieve.cs
public static class Sieve
{
    public static int[] Primes(int limit)
    {
        List<int> primes = [];

        for (var i = 2; i <= limit; i++)
        {
            if (i == 2)
            {
                primes.Add(i);
                continue;
            }

            if (i % 2 == 0)
            {
                continue;
            }

            var hasDivisor = false;
            for (var x = 3; x <= (int)Math.Sqrt(i); x++)
            {
                if (i % x == 0)
                {
                    hasDivisor = true;
                    break;
                }
            } 

            if (!hasDivisor)
            {
                primes.Add(i);
            }
        }

        return [.. primes];
    }
}

File: cpp/sieve/sieve.cpp
#include "sieve.h"
#include <cmath>

namespace sieve
{

    std::vector<int> primes(int limit)
    {
        std::vector<int> result;

        for (int i = 2; i <= limit; i++)
        {
            if (i == 2)
            {
                result.push_back(i);
                continue;
            }

            if (i % 2 == 0)
            {
                continue;
            }

            bool hasDivisor = false;

            for (int x = 3; x <= (int)sqrt(i); x++)
            {
                if (i % x == 0)
                {
                    hasDivisor = true;
                    break;
                }
            }

            if (!hasDivisor)
            {
                result.push_back(i);
            }
        }

        return result;
    }

} // namespace sieve
